File: api-tsl/csharp/src/model/schemeinformation/TSLSchemeInformationURI.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml;
using tr.gov.tubitak.uekae.esya.api.tsl.model.core;
using tr.gov.tubitak.uekae.esya.api.tsl.util;

namespace tr.gov.tubitak.uekae.esya.api.tsl.model.schemeinformation
{
    public class TSLSchemeInformationURI : BaseElement
    {
        private readonly IList<TSLURI> informationURIs = new List<TSLURI>(); 

        public TSLSchemeInformationURI(XmlDocument document, IList<TSLURI> iInformationURIs) : base(document)
        {
            addLineBreak();
            informationURIs = iInformationURIs;

            for (int i = 0; i < informationURIs.Count; i++)
            {
                mElement.AppendChild(informationURIs[i].Element);
                addLineBreak();
            }
        }

        public TSLSchemeInformationURI(XmlDocument document)
            : base(document)
        {
            addLineBreak();
        }

        public TSLSchemeInformationURI(XmlElement aElement)
            : base(aElement)
        {
            if(mElement.HasChildNodes)
            {
                XmlNodeList childList = aElement.ChildNodes;
                foreach (XmlNode child in childList)
                {
                    informationURIs.Add(new TSLURI((XmlElement)child));
                }
            }
            else
            {
                throw new TSLException(Constants.TAG_SCHEMEINFORMATIONURI + TSL_DIL.NODE_CANNOT_BE_FOUND);
            }
        }

        public  void RemoveAllURIs()
        {
            informationURIs.Clear();
            mElement.RemoveAll();
            addLineBreak();
        }

        public void addInformationURI(TSLURI tslURI)
        {
            informationURIs.Add(tslURI);
            mElement.AppendChild(tslURI.Element);
            addLineBreak();
        }

        public override string LocalName
        {
            get { return Constants.TAG_SCHEMEINFORMATIONURI; }
        }

        public override string Namespace
        {
            get
            {
                return Constants.NS_TSL;
            }
        }

        public IList<TSLURI> InformationURIs
        {
            get { return informationURIs; }
        }

        public TSLURI InformationURIAt(int pos)
        {
            if (pos >= informationURIs.Count)
            {
                return null;
            }
            else
            {
                return informationURIs[pos];
            }
        }

        public string InformationURIStrAt(int pos)
        {
            if (pos >= informationURIs.Count)
            {
                return null;
            }
            else
            {
                return informationURIs[pos].TslURI;
            }
        }
    }
}

File: api-tsl/src/tr/gov/tubitak/uekae/esya/api/tsl/model/schemeinformation/TSLSchemeInformationURI.java
package tr.gov.tubitak.uekae.esya.api.tsl.model.schemeinformation;

import java.util.ArrayList;
import java.util.List;
import org.w3c.dom.Document;
import org.w3c.dom.Element;
import org.w3c.dom.Node;
import org.w3c.dom.NodeList;

import tr.gov.tubitak.uekae.esya.api.tsl.TSLException;
import tr.gov.tubitak.uekae.esya.api.tsl.model.core.BaseElement;
import tr.gov.tubitak.uekae.esya.api.tsl.util.Constants;
import tr.gov.tubitak.uekae.esya.api.tsl.util.TSL_DIL;

public class TSLSchemeInformationURI extends BaseElement {
	private List<TSLURI> informationURIs = new ArrayList<TSLURI>();

	public TSLSchemeInformationURI(Document document,
			List<TSLURI> iInformationURIs) {
		super(document);
		addLineBreak();
		informationURIs = iInformationURIs;

		for (int i = 0; i < informationURIs.size(); i++) {
			mElement.appendChild(informationURIs.get(i).getElement());
			addLineBreak();
		}
	}

	public TSLSchemeInformationURI(Document document) {
		super(document);
		addLineBreak();
	}

	public TSLSchemeInformationURI(Element aElement) throws TSLException {
		super(aElement);
		if (mElement.hasChildNodes()) {
			NodeList childList = aElement.getElementsByTagName("*");
			for (int i = 0; i < childList.getLength(); i++) {
				Node child = childList.item(i);
				informationURIs.add(new TSLURI((Element) child));
			}
		} else {
			throw new TSLException(Constants.TAG_SCHEMEINFORMATIONURI
					+ TSL_DIL.NODE_CANNOT_BE_FOUND);
		}
	}

	public void RemoveAllURIs() {
		informationURIs.clear();
		RemoveAll();// mElement.RemoveAll();
		addLineBreak();
	}

	public void addInformationURI(TSLURI tslURI) {
		informationURIs.add(tslURI);
		mElement.appendChild(tslURI.getElement());
		addLineBreak();
	}

	@Override
	public String getLocalName() {
		return Constants.TAG_SCHEMEINFORMATIONURI;
	}

	@Override
	public String getNamespace() {

		return Constants.NS_TSL;

	}

	public List<TSLURI> getInformationURIs() {
		return informationURIs;
	}

	public TSLURI InformationURIAt(int pos) {
		if (pos >= informationURIs.size()) {
			return null;
		} else {
			return informationURIs.get(pos);
		}
	}

	public String InformationURIStrAt(int pos) {
		if (pos >= informationURIs.size()) {
			return null;
		} else {
			return informationURIs.get(pos).getTslURI();
		}
	}
}
